### DISCUSSION TITLE
Does the class returned by System.Security.Cryptography.Aes.Create() uses a managed or a native crypto implementation?

### DISCUSSION THREAD

**Discussion by arendigossweiler** [dotnet/runtime#81155] [Q&A] [answered]
I've developed an app with .NET 7 / Maui which uses encryption.
The app is built for Android and iOS.

I need to know if the class returned by System.Security.Cryptography.Aes.Create() uses a managed or a native crypto implementation before I can deploy the app to the Apple AppStore.

From appstoreconnect.apple.com:
"_You're required to provide documentation if your app contains any of the following:
Standard encryption algorithms instead of, or in addition to, using or accessing the encryption within Apple's operating system_"

I do not need to provide the export compliance documentation if my app uses native iOS encryption.

**Comment by bartonjs** [comment, marked as answer]
Native.  There is no managed implementation of AES in .NET Core / .NET 5+.  (The type AesManaged exists, but even it uses native libraries on .NET Core / .NET 5+)

**Reply by Fabi** [reply]
Sadly that's the case :(

**Reply by bartonjs** [reply]
@Fabi why "sadly"?  Except for extremely small payloads, the native versions are faster than AesManaged was (because it was never written to take advantage of aes-ni on processors that supported it), and the managed versions were/would-be a compliance burden (e.g. they can't be used by anyone subject to FIPS/FedRAMP, and they create an import/export burden for self-contained applications).

There's basically no advantage to anyone other than the framework team for a managed implementation... and that advantage would be we get to write it once and not need to learn how to do it on new OSes.  But since we do it in the core layer, applications still get the "it works everywhere" benefit without the compliance hit.
